OSDOCS-1623 Support for AWS C2S secret region (#28367)
openshift/openshift-docs: modules/installation-aws-about-government-region.adoc

* `us-gov-west-1`
* `us-gov-east-1`

The following AWS Secret Region partition is supported:

* `us-iso-east-1`

The AWS government or secret region, and accompanying custom AMI, must be manually configured in the
`install-config.yaml` file since {op-system} AMIs are not provided by Red Hat
for those regions.

[IMPORTANT]
====

> If you are deploying to the C2S Secret Region, you must also define a custom CA certificate in the `additionalTrustBundle` field of the `install-config.yaml` file, since the AWS API requires a custom CA trust bundle. 

Reviewer: @codyhoag , besides `additionalTrustBundle` field in install-config.yaml, AWS_CA_BUNDLE env is also supported, please @staebler confirm, thanks.

Reviewer 2: No, that is not correct. Setting the AWS_CA_BUNDLE env variable will only affect interactions that the installer has with the AWS API. It will not affect the behavior of the cluster. The user **must** use the `additionalTrustBundle` field.

Note, however, that the installer must also be able to access the AWS API, so the CA bundles need to also be set up on the machine running the installer. This can be done by adding the CA bundle to the trust store on the machine or by using the AWS_CA_BUNDLE env variable.

Author: @staebler do you think appending a statement to the current admonition would be beneficial? Perhaps something like this?:
>To allow the installation program to access the AWS API, the CA certificates must also be defined on the machine running the installation program. This can be done by adding the CA bundle to the trust store on the machine or by using the `AWS_CA_BUNDLE` environment variable.

Reviewer 2: Sure, I think there is value in that. Apparently, you can also set the CA bundle to use in your aws config file by setting the ca_bundle field [1].

[1] https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-ca_bundle.html

Author: @yunjiang29 I added additional info for this in 793f617d02f7d668f0a4295e00974db67f69b13d. Please let me know if you have any further suggestions.

> If you are deploying to the C2S Secret Region, you must also define a custom CA certificate in the `additionalTrustBundle` field of the `install-config.yaml` file, since the AWS API requires a custom CA trust bundle. To allow the installation program to access the AWS API, the CA certificates must also be defined on the machine running the installation program. This can be done by adding the CA bundle to the trust store on the machine, using the `AWS_CA_BUNDLE` environment variable, or defining the CA bundle in the link:https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-ca_bundle.html[`ca_bundle`] field of the AWS config file.

Reviewer 3: ```suggestion
If you are deploying to the C2S Secret Region, you must also define a custom CA certificate in the `additionalTrustBundle` field of the `install-config.yaml` file because the AWS API requires a custom CA trust bundle. To allow the installation program to access the AWS API, the CA certificates must also be defined on the machine that runs the installation program. You must add the CA bundle to the trust store on the machine by defining the `AWS_CA_BUNDLE` environment variable or defining the CA bundle in the link:https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-ca_bundle.html[`ca_bundle`] field of the AWS config file.
```
